az-hadoop-jobtype-plugin/src/main/java/azkaban/jobtype/HadoopJavaJobRunnerMain.java

```diff
@@ -361,60 +354,52 @@ private static Object getObject(String jobName, String className,
 
       if (propsClass != null
           && ((getConstructor(runningClass, String.class, propsClass) != null)
-          || (getConstructor(runningClass, String.class, propsClass, propsClass, loggerClass) != null))) {
+          || (getConstructor(runningClass, String.class, propsClass, propsClass) != null))) {
         // is this the props class
         break;
       }
       propsClass = null;
     }
 
     Object obj = null;
-    if (propsClass != null && loggerClass != null &&
-        getConstructor(runningClass, String.class, propsClass, propsClass, loggerClass) != null) {
-      // Handle cases where jobtype has ctor of format
-      // (java.lang.String,azkaban.utils.Props,azkaban.utils.Props,org.apache.log4j.Logger)
-      logger.info("Creating object for ctor(java.lang.String,azkaban.utils.Props,azkaban.utils"
-          + ".Props,org.apache.log4j.Logger)");
-      // Create SysProps class
-      Constructor<?> sysPropsCon =
-          getConstructor(propsClass, propsClass, Properties[].class);
-      Object sysProps =
-          sysPropsCon.newInstance(null, new Properties[]{properties});
+    if (propsClass != null) {
       // Create jobProps class
       Constructor<?> jobPropsCon =
           getConstructor(propsClass, propsClass, Properties[].class);
       Object jobProps =
-          jobPropsCon.newInstance(null, new Properties[]{properties});
+          jobPropsCon.newInstance(null, new Properties[]{jobProperties});
 
-      Constructor<?> con =
-          getConstructor(runningClass, String.class, propsClass, propsClass, loggerClass);
-      logger.info("Constructor found " + con.toGenericString());
-      obj = con.newInstance(jobName, sysProps, jobProps, Logger.getLogger(con.getClass()));
-    } else if (propsClass != null
-        && getConstructor(runningClass, String.class, propsClass) != null) {
-      // Create props class
-      Constructor<?> propsCon =
-          getConstructor(propsClass, propsClass, Properties[].class);
-      Object props =
-          propsCon.newInstance(null, new Properties[]{properties});
-
-      Constructor<?> con =
-          getConstructor(runningClass, String.class, propsClass);
-      logger.info("Constructor found " + con.toGenericString());
-      obj = con.newInstance(jobName, props);
+      Constructor<?> con = getConstructor(runningClass, String.class, propsClass);
+      if (con != null) {
+        logger.info("Constructor found " + con.toGenericString());
+        obj = con.newInstance(jobName, jobProps);
+      } else {
+        // Handle cases where jobtype has ctor of format
```

Comment:
Can you replace "ctor" with "constructor"? I think the comment is easier to read that way

Reply:
yes